host/class/uac/usb_host_uac/uac_host.c

```diff
@@ -199,6 +201,33 @@ static esp_err_t _uac_host_device_delete(uac_device_t *uac_device);
 static esp_err_t uac_cs_request_set(uac_device_t *uac_device, const uac_cs_request_t *req);
 static esp_err_t uac_cs_request_set_ep_frequency(uac_iface_t *iface, uint8_t ep_addr, uint32_t freq);
 
+// --------------------------- Utility Functions --------------------------------
+/**
+ * @brief Convert volume U16 type to UAC volume db, for details, please refer to UAC 1.0 spec P.77
+ * 0x7FFF: 127.9961db
+ * 0x0100: 1.0000db
+ * 0x0000: 0.0000db
+ * 0xFF00: -1.0000db (there is a typo in UAC 1.0 spec, it should be 0xFF00)
+ * 0xFE00: -2.0000db
+ * 0x8001: -127.9961db
+ * @param[in] volume      Volume in db
+ * @return float
+ */
+static float _volume_db_i16_2_f(int16_t volume_db)
+{
+    return (float)(volume_db) / 256.0f;
+}
+
+static int16_t _volume_db_f_2_i16(float volume_db_f)
+{
+    if (volume_db_f > 127.9961) {
```

Comment:
Usually, it is a good idea to make all of the constants as a `#define` variable. 
In that case, if you need (just as example) to change the 127.9961 in the code, you will have only one place where you need to do that: `#define VOLUME_MAX_DB (127.9961f)`